### PR TITLE
GH 2393 Move calculationColumns from board to view

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary

Moves calculated columns from board block type to view block type, so that each view can have their own calculations.
Includes migration and removes `column_calculations` from new Board table.

#### Ticket Link

  Fixes https://github.com/mattermost/focalboard/issues/2695
  and 
  Fixes https://github.com/mattermost/focalboard/issues/2393

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM 👍

**Review by harshilsharma63 — APPROVED** [review]
LGTM 👍

File: server/services/store/sqlstore/migrations/000017_add_teams_and_boards.up.sql [hunk 1]
```diff
@@ -11,16 +11,42 @@ ALTER TABLE {{.prefix}}blocks ADD COLUMN board_id VARCHAR(36);
...
 {{- /* cleanup incorrect data format in column calculations */ -}}
+{{- /* then move from 'board' type to 'view' type*/ -}}
 {{if .mysql}}
-UPDATE {{.prefix}}blocks SET fields = JSON_SET(fields, '$.columnCalculations', cast('{}' as json))  WHERE fields->'$.columnCalculations' = cast('[]' as json);
+UPDATE {{.prefix}}blocks SET fields = JSON_SET(fields, '$.columnCalculations', cast('{}' as json)) WHERE fields->'$.columnCalculations' = cast('[]' as json);
+
+UPDATE {{.prefix}}blocks b
+  JOIN (
+    SELECT id, fields->'$.columnCalculations' as board_calculations from {{.prefix}}blocks
+    WHERE fields -> '$.columnCalculations' <> cast('{}' as json) and fields -> '$.columnCalculations' <> cast('[]' as json)
```
Comment (mgdelacroix) [inline server/services/store/sqlstore/migrations/000017_add_teams_and_boards.up.sql:21]: nit: we don't need to consider `[]` here as it has just been replaced in the previous `UPDATE`

```suggestion
    WHERE fields -> '$.columnCalculations' <> cast('{}' as json)
```

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1: left a minor, non-blocking nit

**Pull request merged**